Documentation/botr/ryujit-overview.md

```diff
@@ -103,7 +103,7 @@ The liveness analysis determines the set of defs, as well as the uses that are u
 
 * The live-in and live-out sets are captured in the `bbLiveIn` and `bbLiveOut` fields of the `BasicBlock`.
 * The `GTF_VAR_DEF` flag is set on a lclVar node (all of which are of type `GenTreeLclVarCommon`) that is a definition.
-* The `GTF_VAR_USEASG` flag is set (in addition to the `GTF_VAR_DEF` flag) for the target of an update (e.g. +=).
+* The `GTF_VAR_USEASG` flag is set (in addition to the `GTF_VAR_DEF` flag) on `GT_LCL_FLD` nodes that are partial definitions of a local variable.
```

Comment:
I believe that this is correct, but I think it would be clearer to say:

    The `GTF_VAR_USEASG` flag is set (in addition to the `GTF_VAR_DEF` flag) on partial definitions of a local variable (i.e. `GT_LCL_FLD` nodes that do not define the entire variable).

Just to make it clear that the meaning is that it's a partial definition, not that it is specifically a `GT_LCL_FLD` node - those are just the only nodes that are partial definitions.

Thanks!

Reply:
Thanks, updated. Do you think that this flag will ever be used on nodes other than `GT_LCL_FLD`? It seems a bit strange, I'd expect `GT_LCL_VAR` to always be a full def and I don't see what other lcl var nodes could reasonably use this flag.

Reply:
I don't foresee ever using this anywhere bug `GT_LCL_FLD` - my point was just that the fundamental characteristic is not that it's a `GT_LCL_FLD` but that it's a partial def of a lclVar, which, as it happens, is only ever `GT_LCL_FLD` or `GT_STORE_LCL_FLD`.